apm-lambda-extension/apmproxy/apmserver.go

```diff
@@ -43,6 +43,12 @@ func (c *Client) ForwardApmData(ctx context.Context, metadataContainer *Metadata
 			c.logger.Debug("Invocation context cancelled, not processing any more agent data")
 			return nil
 		case agentData := <-c.DataChannel:
+			if agentData.Flushed {
+				c.flushMutex.Lock()
+				c.flushCount--
+				c.flushMutex.Unlock()
+				continue
```

Comment:
Why are we continuing here? IIANM, the `?flushed` request may contain event data.

Reply:
Good catch! Fixed